pkg/clusterresource/controller.go

```diff
@@ -285,18 +283,24 @@ func (c *controller) syncNamespace(ctx context.Context, namespace NamespaceName,
 		}
 		for _, target := range c.executionCluster.GetAllValidTargets() {
 			k8sObjCopy := k8sObj.DeepCopyObject()
+			logger.Debugf(ctx, "Attempting to create resource [%+v](%+v) in cluster [%v] for namespace [%s]",
+				k8sObj.GetObjectKind().GroupVersionKind().Kind, k8sObj, target.ID, namespace)
 			err = target.Client.Create(ctx, k8sObjCopy)
 			if err != nil {
 				if k8serrors.IsAlreadyExists(err) {
 					logger.Debugf(ctx, "Resource [%+v] in namespace [%s] already exists - attempting update instead",
 						k8sObj.GetObjectKind().GroupVersionKind().Kind, namespace)
 					c.metrics.AppliedTemplateExists.Inc()
-					err = target.Client.Patch(ctx, k8sObjCopy, client.MergeFrom(k8sObjCopy))
+					// Use a strategic-merge-patch to mimic `kubectl apply` behavior.
+					err = target.Client.Patch(ctx, k8sObjCopy, StrategicMergeFrom(k8sObjCopy))
```

Comment:
Can you copy/paste maybe the PR description here as well?  And also explain the issue we were seeing before?